sleap_nn/model_trainer.py

```diff
@@ -239,14 +289,27 @@ def __init__(self, config: OmegaConf):
             backbone_config=self.model_config.backbone_config,
             head_configs=[self.model_config.head_configs],
         ).to(self.m_device)
-
-        # Initializing the model weights
-        if self.model_config.init_weights == "xavier":
-            self.model.apply(xavier_init_weights)
         self.training_loss = {}
         self.val_loss = {}
         self.learning_rate = {}
 
+        if self.model_config.init_weights == "xavier":
+            self.model.apply(xavier_init_weights)
+
+        if self.model_config.pre_trained_weights:
+            self._init_pretrain_model()
+
```

Comment:
If both of these configs are set, then the pretrained weights takes precedence. If that's the case, then this should be an if-elif block, and the order of precedence should be documented.